### ISSUE TITLE
[Accessibility]: Code monitoring logs should use list elements

### ISSUE THREAD

**Issue by lguychard** [sourcegraph/sourcegraph#35151]
### Audit type

ARC Toolkit

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34194

### Problem description

1. Navigate to https://sourcegraph.com/code-monitoring.
2. Create some code monitors. Wait for some runs to happen so that you can view logs.
3. Navigate to the "Logs" tab.

Code monitoring logs are displayed like a nested collapsible list, but are not implemented using list elements.

### Expected behavior

- Top-level list of code monitors should be an `<ol>`. Timestamp of last run at that level should be right-aligned.
- List of code monitor runs ("Run X days/minutes ago") should be an `<ul>`.
- List of triggers and actions should be an `<ol>`.

### Additional details

<img width="898" alt="Screenshot 2022-05-09 at 15 18 51" src="https://user-images.githubusercontent.com/1741180/167484261-a5773b74-2320-4480-afaf-f363c2030bae.png">

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [X] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by GitStart. I have assigned the label 'gitstart' and added the assignee 'gitstart-sourcegraph'.

**Comment by sourcegraph-bot-2** [comment]
Heads up @benvenker @lguychard - the "team/search-product" label was applied to this issue.
